c/frame/_repr_html_.cc

```diff
@@ -0,0 +1,270 @@
+      html << "</div>";
+    }
+
+    void render_table_header() {
+      const std::vector<std::string>& colnames = dt->get_names();
+      html << "<thead><tr>";
+      html <<   "<td class=row_index></td>";
+      for (size_t j = 0; j < ncols; ++j) {
+        if (j == cols0) {
+          j = ncols - cols1;
+          html << "<th clas=vellipsis>&hellip;</th>";
+        }
+        html << "<th>";
+        render_escaped_string(colnames[j].data(), colnames[j].size());
+        html << "</th>";
+      }
+      html << "</tr></thead>";
+    }
+
+    void render_table_body() {
+      html << "<tbody>";
+      for (size_t i = 0; i < nrows; ++i) {
+        if (i == rows0) {
+          i = nrows - rows1;
+          render_ellipsis_row();
+        }
+        render_data_row(i);
+      }
+      html << "</tbody>";
+    }
+
+    void render_ellipsis_row() {
+      html << "<tr>";
+      html << "<td class=hellipsis>&middot;&middot;&middot;</td>";
+      for (size_t j = 0; j < ncols; ++j) {
+        if (j == cols0) {
+          j = ncols - cols1;
+          html << "<td></td>";
+        }
+        html << "<td class=hellipsis>&middot;&middot;&middot;</td>";
+      }
+      html << "</tr>";
+    }
+
+    void render_data_row(size_t i) {
+      html << "<tr>";
+      html << "<td class=row_index>" << i << "</td>";
+      for (size_t j = 0; j < ncols; ++j) {
+        if (j == cols0) {
+          j = ncols - cols1;
+          html << "<td class=vellipsis>&hellip;</td>";
+        }
+        html << "<td>";
+        const Column* col = dt->columns[j];
+        switch (col->stype()) {
+          case SType::BOOL:
+          case SType::INT8:    render_fw_value<int8_t>(col, i); break;
+          case SType::INT16:   render_fw_value<int16_t>(col, i); break;
+          case SType::INT32:   render_fw_value<int32_t>(col, i); break;
+          case SType::INT64:   render_fw_value<int64_t>(col, i); break;
+          case SType::FLOAT32: render_fw_value<float>(col, i); break;
+          case SType::FLOAT64: render_fw_value<double>(col, i); break;
+          case SType::STR32:   render_str_value<uint32_t>(col, i); break;
+          case SType::STR64:   render_str_value<uint64_t>(col, i); break;
+          default:
+            html << "(unknown stype)";
+        }
+        html << "</td>";
+      }
+      html << "</tr>";
+    }
+
+    void render_frame_dimensions() {
+      html << "<div class=frame_dimensions>";
+      render_comma_separated(nrows);
+      html << " row" << (nrows == 1? "" : "s") << " &times; ";
+      render_comma_separated(ncols);
+      html << " column" << (ncols == 1? "" : "s");
+      html << "</div>";
+    }
+
+
+    void render_escaped_string(const char* ch, size_t len) {
+      size_t maxi = std::min(len, size_t(50));
+      for (size_t i = 0; i < maxi; ++i) {
+        char c = ch[i];
+        if (c == '&') html << "&amp;";
+        else if (c == '<') html << "&lt;";
+        else if (c == '>') html << "&gt;";
+        else html << c;
```

Comment:
I would also replace single quotes `\'` with `&apos;`, and double quotes `\"` with `&quot;`. Btw, what about using `switch`, as `c` is `char`? Should look cleaner.

Reply:
You're right, those 2 characters are recommended to be escaped too. Also maybe any characters with ASCII value < 0x20 (such as `\n`, `\r`, `\t`)? They aren't dangerous per se, but perhaps the user would want to know they're present.

Reply:
Yes, these could also be done just for the user convenience. Quotes are probably more important as they could mess up output significantly.